### ISSUE TITLE
Good job!

### ISSUE THREAD

**Issue by Jibec** [subins2000/kde-weblate#5]
Hello,

I'm the maintainer of https://github.com/yunohost-apps/weblate_ynh you may find some info for your weblate hosting.

I'm also the fedora community manager of https://translate.fedoraproject.org

A KDE asked me info on how to use Weblate for KDE community: https://linuxrocks.online/@carl (I'm also in Mastodon as https://mamot.fr/@jibec )

A demo project exists to allow carl to do whatever test he wanted: https://translate.fedoraproject.org/projects/test-kde/

If somehow I can help the KDE community to have a proper translation tooling, I'll be pleased to help
Note, I think Carl is @ognarb on github.

**Comment by subins2000** [comment]
Hi !

I've met Carl on Mastodon. So, is KDE now looking to make a web tool for localization. I've suggested this [before in the KDE i18n mailing list](https://marc.info/?t=154143956900003&r=1&w=2), but the community is divided. Currently, it's up to the individual teams to decide how their localization toolchain should be.

We, the KDE Malayalam team have been using [Weblate for a while](https://kde.smc.org.in) and so far, so good. But the problem we have is the server resources. Weblate actually require a good amount of RAM and things. So, we only include [some files for localization in the instance](https://kde.smc.org.in).

Currently, KDE Assamese team is also looking into setup a Weblate instance for their localization.

It would be great if there's a better, simpler translation tooling for KDE to onboard new contributors in all languages.

The README is not up-to-date. I'm gonna fix it right now. Also, check out [kde-pos](https://github.com/FOSSersVAST/kde-pos) repo where the PO files are stored before pushing to upstream. The README in that repo is fine, but still not good :D.

Thanks for reaching out :)

**Comment by Jibec** [comment]
If server resource is what you need, the fedora community may help, but you'll have to administrate the server yourself. Shall I ask?

I have no idea what the KDE community decided, I only know about the debates.
I'm unsure pontoon is a good match as it is meant for Mozilla which has a really specific organization (and the right people behind to make it work).

**Comment by subins2000** [comment]
> If server resource is what you need, the fedora community may help, but you'll have to administrate the server yourself. Shall I ask?

That would be great. We can test it with the French localization initially I guess. @ognarb, waiting for your thoughts.

Pontoon was just the initial thought. That thread went on to discuss whether there should be a web tool for localizing.

**Comment by ognarb** [comment]
I just read the linked mailing list and I think the opinion of the veteran translators is very interesting. Basically they need to maintain they current workflow (having offline translations, being to fetch translations fast, ...), so both workflow need to work in simultaneous. This means we will have merge conflicts.

I'm currently a bit overworked with the work on the replacement of KDE identity and some other website (other parts of the KDE infra need love). So I will be able to work on it only in a few months. There was a few others in the KDE french community, who were interested in online translations. Maybe they can help more than me in the meantime, otherwise I will work on it after finishing the other things.

**Comment by comradekingu** [comment]
@ognarb It is possible to avoid that, but just pulling the translations from Weblate for offline translation and uploading them there is possible.

**Comment by subins2000** [comment]
@ognarb How about hosting Weblate for small teams ? Let the large teams keep their workflow. Having a Weblate setup will help newcomers (of smaller language teams) to take part easily. Currently, from India, some Assamese & Telugu people have contacted me on how to localize, but the SVN workflow is a bit difficult for a newcomer.
@Jibec @ognarb Would it be possible for KDE to host a Weblate instance for small language teams ? I'll take care of the server setup if it's provided.

@comradekingu Yes, it's possible. Weblate supports downloading PO files (and in other formats), localizing them and upload back. But IIRC only individual uploads is possible, and not batch amount of files.

**Comment by Jibec** [comment]
Le 2020-04-07 13:59, Subin Siby a écrit :
> @Jibec [2] @ognarb [1] Would it be possible for KDE to host a Weblate
> instance for small language teams ? I'll take care of the server setup
> if it's provided.

I'm not part of KDE community, but I'm happy to help.
For the Fedora community, finding for you a server for you to install a 
full Weblate server is doable, but it's not really fast.

But I can welcome you in a few minutes in: 
https://translate.fedoraproject.org

I understand your git repo is: https://github.com/FOSSersVAST/kde-pos
To reduce manual work and allow easy addition of new languages, you'll 
have to commit templates in your git repository: 
https://websvn.kde.org/trunk/l10n-kf5/templates/
Having bot pot and po files, allow Weblate to autodetect new files and 
new language additions.
My suggestion would be to enable the Weblate plugin to update po files 
when pot files gets updated.

Basically, once done, all is automated for git and weblate.
The remaining manual step is you to sync with kde svn.
And if someday you want to add a new language, just include the language 
folder in your git repo. Weblate will detect it.

Would it be ok for you?
If so, just connect there and give me your username, I'll create a 
project kde-l10n and make you admin. You will have freedom to configure 
it the way you like.

Pros:
* great cross project collaboration!
* you'll benefit from our translation memory and contributions
* no server administration for you (you'll jump for Weblate 3.9.1 to 
3.11.3)
* you'll be able to add as many languages you like

Cons:
* Requires your team members to create a Fedora Account
* May be temporary as someday KDE community may have its own platform

**Comment by sripathroy** [comment]
I am part of Telugu Localization Team.  We too were looking to host our localization to Weblate. Most of our contributors are new-comers hence a tool would really help. Our current manual method is consuming time for coordination itself. And tool would help with productivity and inclusion of new comers.

**Comment by subins2000** [comment]
@Jibec I've 

I'm making a new git repo here (to not mess with the already working one) : https://github.com/subins2000/kde-l10n-intermediary

The entire KDE repo has 4435 **template** files. Would Weblate and the server have the capacity to handle this much ? Should I only include particular categories ?

https://websvn.kde.org/trunk/l10n-kf5/templates/messages/

**Comment by Jibec** [comment]
Le 2020-04-08 13:31, Subin Siby a écrit :
> The entire KDE repo has 4435 template files. Would Weblate and the
> server have the capacity to handle this much ? Should I only include
> particular categories ?

Well, it can take it all, don't worry.
We'll create multiple weblate projects for subpart of this work, to make 
it easier for you to follow up (because one file = one weblate 
component)

Please tell me the name of the projects you want me to create, they only 
have to start with "kde-".

**Comment by subins2000** [comment]
@Jibec KDE subgroups POs into `applications`, `kdegames`, `kde-workspace` and so on (45 of em)...

In our [Malayalam localization setup](https://kde.smc.org.in/), we have one project and all files are component with names 'applications/dolphin', `kde-workspace/plasma-nm` etc.
It's easier to manage all components in one project, but 4,000+ components in one project ? It will be difficult to browse through.

For now, let's have one project. Name it `kde-kf5-trunk`.

I'm still wary about whether importing 4000+ PO files would slow the whole instance down :/

**Comment by comradekingu** [comment]
It is possible to have one meta-component, and then all others linked.

**Comment by Jibec** [comment]
I created a project and made you admin: 
https://translate.fedoraproject.org/projects/kde-l10n-intermediary/
The name looks more logical for non kde contributors, and makes it 
explicit it isn't about.

Let's do it, please commit ml and templates in: 
https://github.com/subins2000/kde-l10n-intermediary
(you should be able to keep your git history by adding 
https://github.com/subins2000/kde-l10n-intermediary as a remote and 
force push your work)

I would prefer to handle the first configuration if you don't mind. I 
would like to make sure users can't add language by mistake, and that 
the component discovery addon is properly configured.

**Comment by Jibec** [comment]
In addition, as your git repository is dedicated to weblate, please make 
sure this user has commit rights on your repository: 
https://github.com/weblate
It will allow weblate to do direct commit, without pull requests, saving 
you time.

**Comment by subins2000** [comment]
@Jibec I've pushed the files to the repo (Malayalam, Telugu & Templates). The import command and expression is mentioned here : https://github.com/subins2000/kde-weblate#importing-components-to-weblate

Also, can you please enable suggestions voting mentioned there too

**Comment by Jibec** [comment]
Hello, I have added the applications folder.

Issues to be solved by weblate support: how to limit the list of
languages to make sure we can only translate in ml and te. I sent them
an email a few minutes ago.

I'll also ask weblate support of we can allow translations based on
suggestions only for this project. But I want first to solve the
language limit. I'm afraid of conflict with other KDE language teams.
I've added a global text on project to prevent issues.

In general, no name means no attribution and it creates an
authorship/legal issue, but as you'll do manual commit in SVN, it solves
it, but I can't allow it globally for the whole Fedora translation platform.

Have a nice day,
-- 
Jean-Baptiste Holcroft

**Comment by subins2000** [comment]
> how to limit the list of languages to make sure we can only translate in ml and te

I've mentioned it in [that README](https://github.com/subins2000/kde-weblate#importing-components-to-weblate) :

`UPDATE trans_component SET license='Under the same license as the package', new_lang='none';`

This'll prevent new language additions. But this query needs to be changed to have a WHERE clause to limit it to the project we need.

> I'll also ask weblate support of we can allow translations based on suggestions only for this project

There's this

`weblate shell -c 'from weblate.trans.models import Component; Component.objects.all().update(suggestion_voting=True, suggestion_autoaccept=3, allow_translation_propagation=False)'` 

But it also needs to be modified to limit to the project we need.

> In general, no name means no attribution and it creates an authorship/legal issue

Sorry, but I don't understand this. Do you mean translators won't get credit if a suggestion procedure is followed ?

**Comment by tosky** [comment]
I understand the direction, but I'm sad to see this.

Again and again: there is no opposition in having a global tool on kde.org. But first we need to change something on the infrastructural level (a restructuring connected to the migration to gitlab is one of the many) before adding such a tool (for the languages which require it).
